frontend/catalyst/utils/runtime.py

```diff
@@ -100,4 +103,11 @@ def extract_backend_info(device):
                 device._s3_folder  # pylint: disable=protected-access
             )
 
+    if hasattr(device, "_mcmc"):
+        device_kwargs["mcmc"] = device._mcmc
+    if hasattr(device, "_num_burnin"):
+        device_kwargs["num_burnin"] = device._num_burnin
+    if hasattr(device, "_kernel_name"):
+        device_kwargs["kernel_name"] = device._kernel_name
```

Comment:
Is there a way we can append arbitrary device kwargs that don't rely on us hard coding them here?

Reply:
There is none at the moment. But in a comment (that has since been removed from this PR :sweat_smile: ) I proposed the idea of having a kwargs field in the QNode that just saves all of them and we just forward all of them to the device.

Reply:
:+1: to this comment thread -- part of the motivation behind this feature was to avoid us hardcoding device options in the runtime/frontend, but instead find a way for device developers to define them for use by Catalyst.

E.g., an example of this was the Braket device. We were hardcoding some of the braket device options into the frontend/runtime (e.g., `s3_bucket`), which is not ideal.

@dime10 @erick-xanadu do you have an idea of how we can avoid the hardcoding?

Reply:
I think we cannot get around some things. The frontend should just store all the information it wants in a dictionary and then the frontend can just encode this in the IR as we do so right now. The runtime definitely needs to forward these arguments to the device. And the device then will set up whatever it needs with these variables.

Having the options specified in the toml file is just a bit of indirection. In the frontend, I will need to read the toml file, then access the device fields based on those strings, and finally store them in a dictionary which I will finally be encoded in the IR (as we do so right now). I would prefer if the device implementors (of `qml.Device`) would just store all the arguments that they want forwarded to the device ( implementation of `class QuantumDevice`). The assumption is that these people implementing both of these are likely the same people and therefore if they change something on `class QuantumDevice`, let's say a parameter name, then they just modify the frontend to pass that parameter name as a key in the dictionary I am suggesting instead of the old key.

The above is for runtime options, but a similar approach should be good for compile time options. (We don't really have many compile time options at the moment, but once we have...) the implementors of `qml.Device` can just specify compile time options in a dictionary and we forward these to whatever compiler we are using. Please note that this assumes the possibility that each `qml.Device` and `class QuantumDevice` may have their own compiler, which I believe is entirely possible, but based on our conversations we are assuming that is not the case and the compiler is known at the `qjit` level so the compiler options should live on the `qjit` decorator.

Reply:
Yes I think having a specific field in the `qml.Device` class for `runtime_options` that returns the dictionary of all the options it wants forwarded to the device initializer would be best. That would require PRs to PennyLane and the device plugins (e.g. Amazon Braket).
In the absence of that, or until then, I think specifying the list of properties to fetch from the device object in the TOML file would still be preferable to hardcoding them in the frontend for each device.

Reply:
Let me add this to the backlog :)